### ISSUE TITLE
Add support for HTTP basic auth

### ISSUE THREAD

**Issue by chrishiestand** [stianeikeland/node-etcd#41]
HTTP Basic Auth support was added in etcd 2.1, so this library should support it.

**Comment by stianeikeland** [comment]
I agree! 
Sorry for the slow reply, been away on vacation, will try to get on this soon.

**Comment by chrishiestand** [comment]
No problem. Thanks @stianeikeland

**Comment by gronnbeck** [comment]
Is Basic Auth going to be supported any day soon?

**Comment by stianeikeland** [comment]
Finally got some time to work on this (urls-and-basic-auth branch).
I'm going to restructure things a bit and release a new major version, constructor will end up looking like something close to this:

```
options = { 
  auth: { username: "user", password: "pass },
  ssl: { certs... },
  ....
}
etcd = new Etcd(["localhost:4001", "otherhost:4001", ....], options);
```

**Comment by obezuk** [comment]
+1

**Comment by lorenz** [comment]
+1

**Comment by stianeikeland** [comment]
Could you try the beta release? Published as node-etcd@beta
```$ npm install node-etcd@beta```

https://github.com/stianeikeland/node-etcd#basic-auth
Note that constructor has changed slightly.

And sorry for the delay here, life got in the way.
Thank you! :)

**Comment by stianeikeland** [comment]
Was released some time ago as 5.0.1, forgot to close.
